src/components/structures/MessagePanel.js

```diff
@@ -238,26 +239,41 @@ module.exports = React.createClass({
         return !this._isMounted;
     },
 
-    _getEventTiles: function() {
+    // TODO: Implement granular (per-room) hide options
+    _shouldShowEvent: function(mxEv) {
+        // console.log("_shouldShowEvent ", mxEv.getId(), mxEv);
         const EventTile = sdk.getComponent('rooms.EventTile');
+        if (!EventTile.haveTileForEvent(mxEv)) {
+            return false; // no tile = no show
+        }
+
+        // Always show highlighted event
+        if (this.props.highlightedEventId === mxEv.getId()) return true;
+
+        return !shouldHideEvent(mxEv, this._syncedSettings);
+    },
+
+    _getEventTiles: function() {
         const DateSeparator = sdk.getComponent('messages.DateSeparator');
         const MemberEventListSummary = sdk.getComponent('views.elements.MemberEventListSummary');
 
+        let visible = this.props.readMarkerVisible;
```

Comment:
This fixes implicit var hoisting which may/may not cause a bug if MELS is the first thing to be rendered. https://matrix.to/#/!DdJkzRliezrwpNebLk:matrix.org/$1499432050320hAocg:ldbco.de